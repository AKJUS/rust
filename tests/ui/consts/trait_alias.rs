#![feature(trait_alias, const_trait_impl)]
//@ revisions: pass fail

const trait Bar {
    fn bar(&self) {}
}
const trait Baz {
    fn baz(&self) {}
}

impl const Bar for () {}
impl const Baz for () {}

const trait Foo = [const] Bar + Baz;

const fn foo<T: [const] Foo>(x: &T) {
    x.bar();
    #[cfg(fail)]
    {
        x.baz();
        //[fail]~^ ERROR: the trait bound `T: [const] Baz` is not satisfied
    }
}

const _: () = foo(&());
//~^ ERROR: `(): const Foo` is not satisfied

fn main() {}
