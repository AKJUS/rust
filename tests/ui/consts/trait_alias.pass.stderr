error[E0277]: the trait bound `(): const Foo` is not satisfied
  --> $DIR/trait_alias.rs:28:19
   |
LL | const _: () = foo(&());
   |               --- ^^^
   |               |
   |               required by a bound introduced by this call
   |
note: required by a bound in `foo`
  --> $DIR/trait_alias.rs:19:17
   |
LL | const fn foo<T: [const] Foo>(x: &T) {
   |                 ^^^^^^^^^^^ required by this bound in `foo`

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0277`.
